PictureApp.API.Services/FollowerService.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Security.Authentication;
+using System.Threading.Tasks;
+using AutoMapper;
+using PictureApp.API.Data;
+using PictureApp.API.Data.Repositories;
+using PictureApp.API.Dtos;
+using PictureApp.API.Extensions.Exceptions;
+using PictureApp.API.Models;
+using PictureApp.API.Services;
+
+namespace PictureApp.API.Services
+{
+    public class FollowerService : IFollowerService
+    {
+        private readonly IUserService _userService;
+        private readonly IRepository<UserFollower> _userFollowerRepository;
+        private readonly IRepository<User> _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public FollowerService(
+            IUserService userService,
+            IRepository<UserFollower> userFollowerRepository, 
+            IRepository<User> userRepository, 
```

Comment:
Do we really need both IRepository<User> and IUserService dependencies, whilst they are doing almost the same?

Reply:
yes i think, we need functionality for get user by user Id from User repo and we should have save functionality for UserFollow entity from dedicated repository defined by specific repository.

Reply:
Apologize for not being carefully when typing my comment :) I meant IRepository<User> and IUserRepository. In my view, they are very similar so we can use one of them instead both.